src/osp/Active/SysWire.h

```diff
@@ -116,261 +127,488 @@ namespace wiretype
+
+            // Get machine entity from vehicle
+            auto &machines = rScene.reg_get<ACompMachines>(partEnt);
+            ActiveEnt machEnt = machines.m_machines[bpLink.m_protoMachineIndex];
+
+            // Get panel to connect to
+            auto& panel = rScene.reg_get< ACompWirePanel<WIRETYPE_T> >(machEnt);
+
+            // Link them
+            SysWire::connect<WIRETYPE_T>(node, nodeIndex, panel, machEnt,
+                                         portindex_t<WIRETYPE_T>(bpLink.m_port),
+                                         typename WIRETYPE_T::LinkState{});
+        }
     }
 }
-//-----------------------------------------------------------------------------
 
-template<typename T>
-const T* WireInput::get_if() const
+template<typename WIRETYPE_T>
+void SysSignal<WIRETYPE_T>::signal_update_construct(ActiveScene& rScene)
 {
-    if (list() == nullptr)
-    {
-        // Not connected to any WireOutput!
-        return nullptr;
-    }
-    else
+    auto view = rScene.get_registry()
+            .view<osp::active::ACompVehicle,
+                  osp::active::ACompVehicleInConstruction>();
+
+    wire_id_t const id = wiretype_id<WIRETYPE_T>();
+
+    for (auto [vehEnt, rVeh, rVehConstr] : view.each())
     {
-        // Attempt to get value of variant
-        return std::get_if<T>(&(list()->value()));
+        SysWire::construct_panels<WIRETYPE_T>(
+                    rScene, vehEnt, rVeh, *(rVehConstr.m_blueprint));
+
+        signal_construct_nodes(rScene, vehEnt, rVeh, *(rVehConstr.m_blueprint));
     }
 }
 
-inline WireInput::WireInput(IWireElement* element, std::string name) noexcept
- : m_element(element)
- , m_name(std::move(name))
-{ }
+template<typename WIRETYPE_T>
+void SysSignal<WIRETYPE_T>::signal_update_nodes(ActiveScene& rScene)
+{
+    auto &rNodes = rScene.reg_get< ACompWireNodes<WIRETYPE_T> >(rScene.hier_get_root());
+    auto &rWire = rScene.reg_get<ACompWire>(rScene.hier_get_root());
 
-inline WireInput::WireInput(IWireElement *element, WireInput&& move) noexcept
- : LinkedListItem<WireInput, WireOutput>(std::move(move))
- , m_element(element)
- , m_name(std::move(move.m_name))
-{ }
+    // Machines to update accumulated throughout updating nodes
+    std::vector<std::vector<ActiveEnt>> machToUpdate(rWire.m_entToCalculate.size());
 
-//-----------------------------------------------------------------------------
+    // Loop through all nodes that need to be updated
+    for (nodeindex_t<WIRETYPE_T> nodeIndex : rNodes.m_needUpdate)
+    {
+        WireNode<WIRETYPE_T> &rNode = rNodes.get_node(nodeIndex);
+
+        // Overwrite value with value to write
+        rNode.m_state.m_value = rNode.m_state.m_write;
+
+        // Add connected machine inputs to machToUpdate vectors, as they now
+        // require a calculation update
+        for (auto it = std::next(std::begin(rNode.m_links));
+             it != std::end(rNode.m_links); it++)
```

Comment:
if you're going to use std::next and std::begin, then use std::advance(it) instead of it++.